### PR TITLE
add job for reminder email 43

### PR DESCRIPTION
**Pull request by gidsg**
https://trello.com/c/lHm5E9HE/243-2-jenkinsify-please-sign-your-framework-agreement-script-reminder
add job for reminder email 43
I have tested this https://ci.marketplace.team/job/remind-suppliers-to-sign-framework-agreement-(43)/1/consoleFull

### CODE REVIEW

File: job_definitions/notify-suppliers-reminder-sign-agreement.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            - staging
+            - production
+      - string:
+          name: FRAMEWORK_SLUG
+          default: null
+          description: "Framework Slug e.g. 'digital-outcomes-and-specialists-5'"
+      - string:
+          name: SUPPLIER_IDS
+          default: null
+          description: "Comma-separated list of Supplier IDs to be emailed (e.g. '1234,5678')"
```
Comment (bjgill) [inline job_definitions/notify-suppliers-reminder-sign-agreement.yml:20]: Is it obvious to someone using this job what happens if this is left empty?

File: job_definitions/notify-suppliers-reminder-sign-agreement.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                if [ -n "$SUPPLIER_IDS" ]; then
+                  FLAGS="$FLAGS --supplier-id=$SUPPLIER_IDS"
+                fi
+
+                if [ "$DRY_RUN" = "true" ]; then
+                  FLAGS="$FLAGS --dry-run"
+                fi
+
+                docker run --rm \
+                -e DM_DATA_API_TOKEN_${STAGE^^} \
```
Comment (bjgill) [inline job_definitions/notify-suppliers-reminder-sign-agreement.yml:45]: That's really neat - is this a bash thing?

**Review by bjgill — COMMENTED** [review]
Looks sensible. Just one substantive question about documentation.

File: job_definitions/notify-suppliers-reminder-sign-agreement.yml (hunk 2 above)
Comment (gidsg) [reply job_definitions/notify-suppliers-reminder-sign-agreement.yml:45]: yes, apparently this was added in Bash 5 https://www.gnu.org/software/bash/manual/bash.html#index-parameter-expansion described as ``${parameter^^pattern}``

File: job_definitions/notify-suppliers-reminder-sign-agreement.yml (hunk 1 above)
Comment (gidsg) [reply job_definitions/notify-suppliers-reminder-sign-agreement.yml:20]: Good point. I can update the description to make this clearer

Comment (gidsg) [reply job_definitions/notify-suppliers-reminder-sign-agreement.yml]: added in 38c3195

**Review by bjgill — APPROVED** [review]
:tada:

**Pull request merged**